service/service.go

```diff
@@ -71,6 +71,19 @@ func newService(set *settings) (*service, error) {
 	if err = srv.telemetryInitializer.init(set.BuildInfo, srv.telemetrySettings.Logger, set.Config.Service.Telemetry, set.AsyncErrorChannel); err != nil {
 		return nil, fmt.Errorf("failed to initialize telemetry: %w", err)
 	}
+
+	// Used to capture main error and any defer errors
+	var returnErrs error
+
+	// If there is any error after we initialize telemetry we need to ensure we clean that up for any future service
+	defer func() {
+		if returnErrs != nil {
+			if shutdownErr := srv.telemetryInitializer.shutdown(); shutdownErr != nil {
+				returnErrs = multierr.Append(returnErrs, fmt.Errorf("failed to shutdown collector telemetry: %w", shutdownErr))
+			}
+		}
+	}()
```

Comment:
Business logic aside, we should flatten this by returning early in the simple case.
```suggestion
	defer func() {
		if returnErrs == nil {
			return
		}
		if shutdownErr := srv.telemetryInitializer.shutdown(); shutdownErr != nil {
			returnErrs = multierr.Append(returnErrs, fmt.Errorf("failed to shutdown collector telemetry: %w", shutdownErr))
		}
	}()
```

Reply:
Yeah I agree with this, nested error checking could be cleaned up. I'll make a change.